File: src/public/cognitoAdmin/disableUser.js
"use strict";

const { requireSuperAdmin } = require("../../common/auth");
const { disableUserInCognito } = require("./cognito");

async function handleCognitoDisableUser({ method, res, auth, params }) {
  const okRes = requireSuperAdmin({ auth, res });
  if (okRes !== true) return okRes;

  if (method !== "POST") return res.methodNotAllowed("Method not allowed");

  try {
    const r = await disableUserInCognito(params.username);
    return res.ok({ username: params.username, ...r });
  } catch (e) {
    return res.badRequest("COGNITO_ERROR", e?.message || String(e));
  }
}

module.exports = { handleCognitoDisableUser };

File: src_C/Public/CognitoAdmin/DisableUser.cs
using Amazon.Lambda.APIGatewayEvents;
using RecallSmith.Lambda.Common;

namespace RecallSmith.Lambda.Public.CognitoAdmin;

public static class DisableUser
{
  public static async Task<APIGatewayProxyResponse> HandleCognitoDisableUser(
    LambdaRequest req,
    Res res,
    AuthContext auth,
    IReadOnlyDictionary<string, string> parameters)
  {
    var deny = Auth.RequireSuperAdmin(auth, res);
    if (deny is not null) return deny;

    if (req.Method != "POST") return res.MethodNotAllowed("Method not allowed");

    var username = parameters.TryGetValue("username", out var u) ? u : string.Empty;
    try
    {
      var r = await CognitoClient.DisableUserInCognito(username);
      var outObj = new Dictionary<string, object?>(r) { ["username"] = username };
      return res.Ok(outObj);
    }
    catch (Exception ex)
    {
      return res.BadRequest("COGNITO_ERROR", ex.Message);
    }
  }
}

